src/py/flwr/cli/install.py

```diff
@@ -149,21 +158,36 @@ def validate_and_install(
         )
         raise typer.Exit(code=1)
 
-    publisher = config["tool"]["flwr"]["app"]["publisher"]
-    project_name = config["project"]["name"]
-    version = config["project"]["version"]
+    version, fab_id = get_metadata_from_config(config)
+    publisher, project_name = fab_id.split("/")
 
-    if (
-        fab_name
-        and fab_name != f"{publisher}.{project_name}.{version.replace('.', '-')}"
-    ):
-        typer.secho(
-            "❌ FAB file has incorrect name. The file name must follow the format "
-            "`<publisher>.<project_name>.<version>.fab`.",
-            fg=typer.colors.RED,
-            bold=True,
+    if fab_name:
+        fab_publisher, fab_project_name, fab_version, fab_shorthash = (
+            get_metadata_from_fab_filename(fab_name)
         )
-        raise typer.Exit(code=1)
+        if (
+            f"{fab_publisher}.{fab_project_name}.{fab_version}"
+            != f"{publisher}.{project_name}.{version}"
+            or len(fab_shorthash) != FAB_HASH_TRUNCATION  # Verify hash length
+        ):
+
+            typer.secho(
+                "❌ FAB file has incorrect name. The file name must follow the format "
+                "`<publisher>.<project_name>.<version>.<8hexchars>.fab`.",
+                fg=typer.colors.RED,
+                bold=True,
+            )
+            raise typer.Exit(code=1)
+
+        try:
+            _ = int(fab_shorthash, 16)  # Verify hash is a valid hexadecimal
+        except ValueError as e:
+            typer.secho(
+                f"❌ FAB file has an invalid hexadecimal string `{fab_shorthash}`.",
+                fg=typer.colors.RED,
+                bold=True,
+            )
+            raise typer.Exit(code=1) from e
```

Comment:
What about moving this to the `get_metadata_from_fab_filename` function? And have it return None if there is an error

Reply:
Good point. I've refactored it to a `_validate_fab_and_config_metadata()`.